features/steps/steps.py

```diff
@@ -93,6 +94,23 @@ def cleanup_instance() -> None:
     context.add_cleanup(cleanup_instance)
 
 
+@when("I run `{command}` {user_spec}, retrying exit [{exit_codes}]")
+def when_i_retry_run_command(context, command, user_spec, exit_codes):
+    when_i_run_command(context, command, user_spec, verify_return=False)
+    retries = [5, 5, 10]  # Sleep times to wait between retries
+    while context.process.returncode not in exit_codes.split(","):
+        if not retries:
+            break
+        print(
+            "--- Retrying on exit {exit_code}: {command}".format(
+                exit_code=context.process.returncode, command=command
+            )
+        )
+        time.sleep(retries.pop(0))
```

Comment:
The error here that will abort the execution will be something like:
```
IndexError: pop from empty list
```

However, I think we should better inform that the command failed after all of the retries attempts

Reply:
Added a warning "Exhausted retries waiting for edit_codes: x,y" and wrapped in a try/except block